python-lib/ontology_tagger.py

```diff
@@ -1,17 +1,35 @@
-"""Main module to tag the documents"""
 from spacy_tokenizer import MultilingualTokenizer
 from formatter_instanciator import FormatterInstanciator
-from plugin_io_utils import generate_unique, replace_nan_values
+from plugin_io_utils import generate_unique, get_keyword
 from spacy.matcher import PhraseMatcher
 from spacy.tokens import Doc
 from fastcore.utils import store_attr
 from typing import AnyStr, List, Union
 import pandas as pd
 from time import perf_counter
 import logging
+from sentence_splitter import SentenceSplitter
 
 
 class Tagger:
+    """Tag text data with a given ontology
+    Relies on spaCy Language components:
+    - Sentencizer to split document into sentences
+    - PhraseMatcher and EntityRuler to tag documents
+    Attributes:
+        ontology_df (pandas dataframe): Ontology which contains the tags to assign
+        tag_column (string): Name of the column in the Ontology. Contains the tags to assign
+        category_column (string): Name of the column in the Ontology. Contains the category of each tag to assign.
+        keyword_column (string): Name of column in the Ontology. Contains the keywords to match with in documents.
+        language (string): Name of the documents language. Used if there is only one language to treat
+        lemmatization (bool): If True, match on lemmatized forms
+        case_insensitivity(bool): If True, match on lowercased forms
+        normalization (bool): If True, normalize diacritic marks e.g., accents, cedillas, tildes
+        tokenizer (MultilingualTokenizer): Tokenizer instance to create the tokenizers for each language
+        matcher_dict (dict): Dictionary of spaCy PhraseMatchers objects. Unused if we are using EntityRuler (in case there are categories in the Ontology)
+        keyword_to_tag(dict): Keywords (key) and tags (value) to retrieve the tags from the matched keywords Unused if we are using EntityRuler (in case there are categories in the Ontology)
```

Comment:
Should this be a private attribute instead of a public one? What is the expected format?

Reply:
Could you explain what do you mean by "what is the expected format" please ? :)

Reply:
Can you give an example of keyword_to_tag?

Reply:
ah yes 👍 7feb6e7

Reply:
Thanks! Nitpick: matcher_dict and keyword_to_tag could be marked as private attributes by naming them _matcher_dict and _keyword_to_tag.

Reply:
here :) 15191ca